package/setup.py

```diff
@@ -131,9 +120,54 @@ def get(self, option_name, default=None):
         except:
             return default
 
+class MDAExtension(Extension, object):
+    """Derived class to cleanly handle setup-time dependencies (numpy)
+    """
+    # The only setup-time numpy dependency comes when setting up its
+    #  include dir.
+    # The actual numpy import and call can be delayed until after pip
+    #  has figured it must install numpy.
+    # This is accomplished by passing the get_numpy_include function
+    #  as one of the include_dirs. This derived Extension class takes
+    #  care of calling it when needed.
+    def __init__(self, *args, **kwargs):
+        try:
+            self._mda_include_dir_args = kwargs.pop("include_dirs")
+        except KeyError:
+            self._mda_include_dir_args = []
+        self._mda_include_dirs = []
+        super(MDAExtension, self).__init__(*args, **kwargs)
+
+    @property
+    def include_dirs(self):
+        if not self._mda_include_dirs:
+            for item in self._mda_include_dir_args:
+                try:
+                    self._mda_include_dirs.append(item()) #The numpy callable
+                except TypeError:
+                    self._mda_include_dirs.append(item) 
+        return self._mda_include_dirs
+
+    @include_dirs.setter
```

Comment:
Is this correct? Setting silently does nothing?  Is this so the property is only created when it is first accessed?